AmplifyPlugins/Core/AWSPluginsCore/Model/Support/SelectionSet.swift

```diff
@@ -53,7 +53,7 @@ extension SelectionSet {
                       let schema = ModelRegistry.modelSchema(from: associatedModelName) {
                 if recursive {
                     var recursive = recursive
-                    if field._isBelongsToOrHasOne {
+                    if field._isBelongsToOrHasOne && !field.isRequired {
```

Comment:
even though it's a minor fix, is there a unit test that can be updated or added to cover this change?